pkg/testresults/results.go

```diff
@@ -0,0 +1,164 @@
+package testresults
+
+import (
+	"encoding/xml"
+	"fmt"
+
+	"github.com/bsm/ginkgo/v2/reporters"
+	"github.com/konflux-ci/qe-tools/pkg/oci"
+	"k8s.io/klog/v2"
+)
+
+// FailureType collects the types of failures met in the PipelineRun
+type FailureType string
+
+const (
+	dropdownSummaryString = "Click to view logs"
+
+	// OtherFailure represents the type failure that hasn't been identified by the analyzer
+	OtherFailure FailureType = "otherFailure"
+	// ClusterCreationFailure represents the issue with provisioning a test cluster
+	ClusterCreationFailure FailureType = "clusterCreationFailure"
+	// TestRunFailure represents the issue with running the test command
+	TestRunFailure FailureType = "testRunFailure"
+	// TestCaseFailure represents the issue in the test suite (i.e. test case has failed)
+	TestCaseFailure FailureType = "testCaseFailure"
+)
+
+// FailedTestCasesReport collects the data about failures
+type FailedTestCasesReport struct {
+	JUnitTestSuites *reporters.JUnitTestSuites
+
+	HeaderString        string
```

Comment:
I don't think this should be apart of the FailedTestCasesReport since it is formatting related.

Reply:
so we chatted and agreed to keep this formatting-related field as a part of `FailedTestCasesReport`